test/unit-tests/decorators.ts

```diff
@@ -0,0 +1,101 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import decoratorsLib = require("../../decorators");
+import yokLib = require("../../yok");
+import chai = require("chai");
+import Promise = require("bluebird");
+import Future = require("fibers/future");
+
+let originalInjector:any = {}
+_.extend(originalInjector, $injector);
+let assert: chai.Assert = chai.assert;
+
+describe("decorators", () => {
+	afterEach(() => {
+		$injector = originalInjector;
+		// Due to bug in lodash's extend method, manually set publicApi to the initial object.
+		$injector.publicApi = {__modules__: {}};
+	});
+
+	describe("exported", () => {
+		it("returns function", () => {
+			let result: any = decoratorsLib.exported("test");
+			assert.equal(typeof(result), "function");
+		});
+
+		it("does not change original method", () => {
+			let testInjector = new yokLib.Yok();
+			let promisifiedResult: any = decoratorsLib.exported("moduleName");
+			let expectedResult = {"originalObject": "originalValue"};
+			let actualResult = promisifiedResult({}, "myTest1", expectedResult);
+			assert.deepEqual(actualResult, expectedResult);
+		});
+
+		it("adds method to public api", () => {
+			let testInjector = new yokLib.Yok();
+			assert.deepEqual($injector.publicApi.__modules__["moduleName"], undefined);
+			let promisifiedResult: any = decoratorsLib.exported("moduleName");
+			let actualResult = promisifiedResult({}, "propertyName", {});
+			assert.deepEqual(typeof($injector.publicApi.__modules__["moduleName"]["propertyName"]), "function");
+		});
+
+		it("returns Promise", () => {
+			$injector = new yokLib.Yok();
+			let result = "result";
+			$injector.register("moduleName", {propertyName: () => {return result;}});
+			assert.deepEqual($injector.publicApi.__modules__["moduleName"], undefined);
+			let promisifiedResultFunction: any = decoratorsLib.exported("moduleName");
+			// Call this line in order to generate publicApi and get the real Promise
+			promisifiedResultFunction({}, "propertyName", {});
+			let promise: any = $injector.publicApi.__modules__["moduleName"]["propertyName"]();
+			assert.equal(typeof(promise.then), "function");
+			assert.equal(typeof(promise.catch), "function");
+			assert.deepEqual(promise.value(), result);
```

Comment:
According [promise/A+](https://promisesaplus.com/) spec, neither .catch, nor .value are part of the API. You are checking against bluebird's implementation, which may be OK with the caveat that if we start using native promises or another library, tests would broke.